oga/backend/users/views/views_answer.py

```diff
@@ -12,23 +12,59 @@
 
 @check_login_required
 @check_request
-@require_http_methods(["POST"])
+@require_http_methods(["GET", "POST"])
 @csrf_exempt
-def create_answer(request, question_id):
-    """create_answer api"""
-    req_data = json.loads(request.body.decode())
-    question_type = req_data['question_type']
-    answer_author = get_user(request)
-    answer_content = req_data['answer_content']
+def get_or_create_answer(request, question_or_answer_id):
+    """function for post answer of question_id or get answer of answer_id
+        POST: create_answer api
+        GET: get_answers api"""
+    if request.method == "POST":
+        req_data = json.loads(request.body.decode())
+        question_type = req_data['question_type']
+        answer_author = get_user(request)
+        answer_content = req_data['answer_content']
+        answer = Answer(question=Question.objects.get(id=question_or_answer_id),
+                        author=Profile.objects.get(user=answer_author),
+                        question_type=question_type,
+                        content=answer_content)
+        answer.save()
+        response_dict = {'question_id': answer.question.id,
+                         'author': answer_author.username,
+                         'question_type': answer.question_type,
+                         'answer_content': answer.content,
+                        }
+        return JsonResponse(response_dict, status=200)
+    elif request.method == "GET":
+        ans = Answer.objects.get(id=question_or_answer_id)
+        question = ans.question
+        response_dict = {
+            'id': ans.id,
+            'author': ans.author.user.username,
+            'publish_date_time': ans.publish_date_time,
+            'question_type': ans.question_type,
+            'content': ans.content,
+            'place_name': question.location_id.name,
+        }
+        return JsonResponse(response_dict, safe=False, status=200)
+    else:
+        # should not reach here.
+        return -1
```

Comment:
Since we already have the request method decorator in think this block might be redundant.

Reply:
When I removed that 'else:' pylint complains about it.